sql/src/main/scala/org/datasyslab/geosparksql/utils/DataFrameFactory.scala

```diff
@@ -0,0 +1,19 @@
+package org.datasyslab.geosparksql.utils
+import com.vividsolutions.jts.geom.Geometry
+
+import org.datasyslab.geospark.formatMapper.shapefileParser.ShapefileReader
+import org.datasyslab.geospark.spatialRDD.SpatialRDD
+import scala.collection.JavaConversions._
+
+object DataFrameFactory {
+  def geometryDFFromShapeFile(spark: org.apache.spark.sql.SparkSession, folderPath: String): org.apache.spark.sql.DataFrame ={
+    val srdd = new SpatialRDD[Geometry]
+    srdd.rawSpatialRDD = ShapefileReader.readToGeometryRDD(spark.sparkContext, folderPath)
+    srdd.analyze()
```

Comment:
Hi @tociek , can you please remove srdd.analyze()? I called srdd.analyze() in the unit test just for test purpose. The analyze here doesn't affect anything in DataFrame.

Reply:
Done. Could you explain what analyze() does and why it is useless for DF?